Sources/PCReadCursorDebouncerManager.swift

```diff
@@ -0,0 +1,81 @@
+import Foundation
+
+class PCReadCursorDebouncerManager {
+    private var roomIdsToDebouncers: [Int: PCReadCursorDebouncer] = [:]
+    private weak var currentUser: PCCurrentUser?
+
+    init(currentUser: PCCurrentUser) {
+        self.currentUser = currentUser
+    }
+
+    func set(cursorPosition position: Int, inRoomId roomId: Int, completionHandler: @escaping PCErrorCompletionHandler) {
+        if let debouncer = roomIdsToDebouncers[roomId] {
+            debouncer.set(position: position, completionHandler: completionHandler)
+        } else {
+            let debouncer = PCReadCursorDebouncer(roomId: roomId, currentUser: currentUser)
+            roomIdsToDebouncers[roomId] = debouncer
+            debouncer.set(position: position, completionHandler: completionHandler)
+        }
+    }
+}
+
+class PCReadCursorDebouncer {
+    private var roomId: Int
+    private weak var currentUser: PCCurrentUser?
+    private var interval: TimeInterval
+    private var timer: Timer?
+
+    private var sendReadCursorPayload: (position: Int, completionHandler: PCErrorCompletionHandler)? = nil
+
+    init(roomId: Int, currentUser: PCCurrentUser?, intervalMilliseconds: Int = 500) {
+        self.roomId = roomId
+        self.currentUser = currentUser
+        self.interval = Double(intervalMilliseconds  / 1000)
+    }
+
+    deinit {
+        self.timer?.invalidate()
+        self.timer = nil
+    }
+
+    func set(position: Int, completionHandler: @escaping PCErrorCompletionHandler) {
+        sendReadCursorPayload = (position: position, completionHandler: completionHandler)
```

Comment:
I think the `max` permission should be respected (rather than just the last one). If you're calling `setReadCursor` asynchronously for a load of positions, you want the greatest one to be respected.